tests/unitary/jupyter/test_browser.py

```diff
@@ -205,8 +206,46 @@ def test_browser_rpc_server_error(
     assert str(exc_info.value) == "-32603: server error"
 
 
+def test_browser_rpc_internal_error(mock_callback, env):
+    error = {
+        "code": -32603,
+        "message": "Internal JSON-RPC error.",
+        "data": {
+            "code": 3,
+            "message": "insufficient funds for gas + value. balance: 0, fee: 116, value: 0",
```

Comment:
a comment how this data was generated would be helpful. in both of these cases i think it would be good to add a comment saying we are agnostic to the exact data, we are just testing that we find the key properly.